src/models/document/document.ts

```diff
@@ -325,19 +345,47 @@ export const createDocumentModel = (snapshot?: DocumentModelSnapshotType) => {
     sharedModelManager,
     documentEnv: {}
   };
-  const document = DocumentModel.create(snapshot, fullEnvironment);
-  addDisposer(document, onAction(document, (call) => {
-    if (!document.content || !call.path?.match(/\/content\/tileMap\//)) {
-      return;
+  try {
+    const document = DocumentModel.create(snapshot, fullEnvironment);
+    addDisposer(document, onAction(document, (call) => {
+      if (!document.content || !call.path?.match(/\/content\/tileMap\//)) {
+        return;
+      }
+      const toolTileId = call.path?.match(/\/content\/tileMap\/([^/]*)/)?.[1];
+      if (toolTileId) {
+        const toolTile = document.content.tileMap.get(toolTileId);
+        toolTile?.onTileAction(call);
+      }
+    }));
+    if (document.content) {
+      sharedModelManager.setDocument(document.content);
     }
-    const toolTileId = call.path?.match(/\/content\/tileMap\/([^/]*)/)?.[1];
-    if (toolTileId) {
-      const toolTile = document.content.tileMap.get(toolTileId);
-      toolTile?.onTileAction(call);
+    return document;
+  } catch (e) {
+    // The only time we've seen this error so far is when MST fails to load the content
+    // because it doesn't match the types of the MST models
+    if (!snapshot) {
+      console.error("Empty document failed to be created");
+      throw e;
     }
-  }));
-  if (document.content) {
-    sharedModelManager.setDocument(document.content);
+
+    if (!snapshot.content) {
```

Comment:
Is this case likely temporary? e.g. some fetch likely failed? I'm wondering if it makes sense to show an error message here too.

Reply:
I haven't run into this case in the wild. I'd suspect that if we got a snapshot then we'll also have content. Since I'm not sure under what conditions this case might happen, I think it is better to let the error stop the usual rendering. Otherwise something even more weird might happen than the current "blank right side".  For example perhaps the key of the DocumentModel is corrupted not just the content, so then this might interfere with other documents if we go ahead and try to make a document out of it.